src/threads/table.ts

```diff
@@ -55,11 +56,22 @@ export class ForumTable {
    * Retrieves all the thread event items for a couple participantId+itemId in ascending order.
    * Limit is currently 1MB of data.
    */
-  async getThreadEvents(participantId: string, itemId: string, options: { limit?: number, asc?: boolean } = {}): Promise<ThreadEvent[]> {
+  async getThreadEvents(
+    participantId: string,
+    itemId: string,
+    options: { limit?: number, asc?: boolean, type?: ThreadEvent['type'] } = {},
+  ): Promise<ThreadEvent[]> {
     const threadId = this.getThreadId(participantId, itemId);
     const result = await this.db.query({
       TableName: this.tableName,
-      ExpressionAttributeValues: { ':tid': { S: threadId } },
+      ExpressionAttributeValues: {
+        ':tid': { S: threadId },
+        ...(options.type && { ':type': { S: options.type } }),
+      },
+      ...(options.type && {
+        FilterExpression: '#type = :type',
+        ExpressionAttributeNames: { '#type': 'type' }, // 'type' is a reserved keyword in dynamodb
```

Comment:
The return of the `abc` vs `:abc` vs `#abc`. Same remarks/recommendations as before apply.

Reply:
Btw, I am not sure calling the attribute "type" in an unstructured db is a good idea as it is only for events that you want to use these, so event_type may be more appropriate. But can be left for later.

Reply:
As agreed, I'll merge as is and rename `type` in another PR, see issue #24